src/StdCheats.sol

```diff
@@ -215,13 +215,8 @@ abstract contract StdCheatsSafe {
         vm.assume(!success || abi.decode(returnData, (bool)) == false);
     }
 
-    function assumeNoPrecompiles(address addr) internal virtual {
-        // Assembly required since `block.chainid` was introduced in 0.8.0.
-        uint256 chainId;
-        assembly {
-            chainId := chainid()
-        }
-        assumeNoPrecompiles(addr, chainId);
+    function assumeNoPrecompiles(address addr) internal pure virtual {
+        assumeNoPrecompiles(addr, _pureChainId());
```

Comment:
Ah interesting, thanks for the clarification there. So should we just keep the `pure` version around then since that can be used anywhere the `view` version is used?

Reply:
`_viewChainId` is used to feed it to `_pureChainId`.